backend-shared/config/db/guardrow_test.go

```diff
@@ -0,0 +1,890 @@
+			Expect(err).To(BeNil())
+			Expect(rowsAffected).Should(Equal(1))
+
+			err = dbq.GetManagedEnvironmentById(ctx, &managedEnvironmentfirst)
+			Expect(err).To(BeNil())
+
+			err = dbq.GetManagedEnvironmentById(ctx, &managedEnvironmentsecond)
+			Expect(true).To(Equal(db.IsResultNotFoundError(err)))
+
+		})
+
+		It("Should test guard row against update and delete for operation", func() {
+			err := db.SetupForTestingDBGinkgo()
+			Expect(err).To(BeNil())
+
+			ctx := context.Background()
+			dbq, err := db.NewUnsafePostgresDBQueries(true, true)
+			Expect(err).To(BeNil())
+			defer dbq.CloseDatabase()
+
+			_, _, _, gitopsEngineInstance, _, err := db.CreateSampleData(dbq)
+			Expect(err).To(BeNil())
+			var testClusterUser = &db.ClusterUser{
+				Clusteruser_id: "test-user-1",
+				User_name:      "test-user-1",
+			}
+			operationfirst := db.Operation{
+				Operation_id:            "test-operation-1",
+				Instance_id:             gitopsEngineInstance.Gitopsengineinstance_id,
+				Resource_id:             "test-fake-resource-id",
+				Resource_type:           "GitopsEngineInstance",
+				State:                   db.OperationState_Waiting,
+				Operation_owner_user_id: testClusterUser.Clusteruser_id,
+			}
+			err = dbq.CreateClusterUser(ctx, testClusterUser)
+			Expect(err).To(BeNil())
+
+			err = dbq.CreateOperation(ctx, &operationfirst, operationfirst.Operation_owner_user_id)
+			Expect(err).To(BeNil())
+
+			testClusterUser = &db.ClusterUser{
+				Clusteruser_id: "test-user-2",
+				User_name:      "test-user-2",
+			}
+			operationsecond := db.Operation{
+				Operation_id:            "test-operation-2",
+				Instance_id:             gitopsEngineInstance.Gitopsengineinstance_id,
+				Resource_id:             "test-fake-resource-id",
+				Resource_type:           "GitopsEngineInstance",
+				State:                   db.OperationState_Waiting,
+				Operation_owner_user_id: testClusterUser.Clusteruser_id,
+			}
+			err = dbq.CreateClusterUser(ctx, testClusterUser)
+			Expect(err).To(BeNil())
+
+			err = dbq.CreateOperation(ctx, &operationsecond, operationsecond.Operation_owner_user_id)
+			Expect(err).To(BeNil())
+
+			operationsecond = db.Operation{
+				Operation_id:            "test-operation-2",
+				Instance_id:             gitopsEngineInstance.Gitopsengineinstance_id,
+				Resource_id:             "test-fake-resource-id-update",
+				Resource_type:           "GitopsEngineInstance",
+				State:                   db.OperationState_Waiting,
+				Operation_owner_user_id: testClusterUser.Clusteruser_id,
+				SeqID:                   operationsecond.SeqID,
+				Created_on:              operationsecond.Created_on,
+				Last_state_update:       operationsecond.Last_state_update,
+			}
+
+			err = dbq.UpdateOperation(ctx, &operationsecond)
+			Expect(err).To(BeNil())
+			Expect(operationsecond.Resource_id).Should(Equal("test-fake-resource-id-update"))
```

Comment:
Need to go GetOperation here, for both operationFirst and operationSecond, to make sure that the Go object is up-to-date with the latest state of the database (it is not updated automatically.)